src/components/header/Header.js

```diff
@@ -1,18 +1,39 @@
 import styled from 'styled-components';
 import {Icon} from '@iconify/react';
 import {bindActionCreators} from 'redux';
-import {useDispatch} from 'react-redux';
+import {useDispatch, useSelector} from 'react-redux';
 import {actionCreators} from '../../state';
+import {useEffect} from 'react';
 
 export default function Header() {
   const {setOpenSideNavigation} = bindActionCreators(actionCreators, useDispatch());
+  const sideNavigation = useSelector(state => state.sideNavigation);
+  useEffect(() => {
+    if (sideNavigation.status) {
+      window.addEventListener('click', handleClick, false);
+
+      return () => {
+        window.removeEventListener('click', handleClick, false);
+      };
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [sideNavigation]);
+
+  function handleClick(event) {
+    event.stopPropagation();
+    if (sideNavigation.status && sideNavigation.click) {
+      setOpenSideNavigation(true);
+    } else {
+      setOpenSideNavigation(false);
+    }
+  }
 
   return (
     <StyledHeader>
       <StyledButton
         aria-label={'Side Navigation'}
         onClick={() => {
-          setOpenSideNavigation(true);
+          setOpenSideNavigation(true, true);
```

Comment:
Why you need to put two booleans in the function at this point and above you need just one boolean?

Reply:
when i clicked on the menu the eventListener was always executed immediately and since the value was true it always closed the navigation for me, now i gave it the 2nd parameter optional and only my click on menu it gets 2 trues so it doesn't close that immediately. in the function itself i only ask is ture and ture set a true and so it knows on the next click it can only do false and close the navigation